### PR TITLE
user related db updates for convai_verified attribute

### PR DESCRIPTION
**Pull request by crazylazylife**

### CODE REVIEW

File: db_funcs.py [hunk 1]
```diff
@@ -110,20 +110,28 @@ def get_api_key_info(email : str) -> str:
...
+            convai_ver_status_query = GET_CONVAI_VERIFICATION_STATUS(email)
+            convai_ver_status_query_result = execute_and_return_results(convai_ver_status_query, conn)
         
-            if len(query_results) > 0:
-                api_key = str(query_results[0]['api_key'])
+            if len(api_key_query_results) > 0:
+                user_verification_details['apiKey'] = api_key_query_results[0]['api_key']
+
+            if len(convai_ver_status_query_result) > 0:
+                user_verification_details["convai_verified"] = convai_ver_status_query_result[0]['convai_verified']
```
Comment (DeepankarAcharyya) [inline db_funcs.py:128]: Like the apiKey value is -1 by default, there must be a default convai_verified value with False.
Basically :
user_verification_details = {"apiKey":-1, "convi_verified":False}

File: db_funcs.py [hunk 2]
```diff
@@ -110,20 +110,28 @@ def get_api_key_info(email : str) -> str:
     GET_API_KEY_DETAILS = """ SELECT api_key
                           FROM (SELECT api_key, generation_timestamp AS gt FROM api_map WHERE email = '{}' ORDER BY gt DESC) AS S
                           LIMIT 1; """
-    api_key = "-1"
+    GET_CONVAI_VERIFICATION_STATUS = """ SELECT convai_verified FROM user_details WHERE email ILIKE '{}'; """
```
Comment (DeepankarAcharyya) [inline db_funcs.py:113]: Using a join will be more beneficial in this case based on the email field.

File: db_funcs.py [hunk 3]
```diff
@@ -141,21 +149,23 @@ def user_login(email : str) -> dict:
...
-                api_key['apiKey'] = query_results[0]['api_key']
+            if len(api_key_query_results) > 0:
+                user_verification_details['apiKey'] = api_key_query_results[0]['api_key']
 
         except Exception as e:
             #print(query)
             print("Error in executing the query for user_login : ",e)
     
-    return api_key
+    return user_verification_details
```
Comment (DeepankarAcharyya) [inline db_funcs.py:168]: For this just the variable name is updated?

File: db_funcs.py [hunk 4]
```diff
@@ -1111,3 +1121,31 @@ def update_api_key(email:str, api_key:str)->dict:
...
+
+            cursor_obj = conn.cursor()
+            cursor_obj.execute(query)
+            cursor_obj.close()
+
+            s = "SUCCESS"
+
+        except Exception as e:
+            #print(query)
+            print("Error in executing the query for the following user ",e)
```
Comment (DeepankarAcharyya) [inline db_funcs.py:1148]: Can we update the error message? just including the function name will be fine, helps in converging upon the error

**Review by DeepankarAcharyya — CHANGES_REQUESTED** [review]

**Comment by DeepankarAcharyya** [comment]
Also please resolve the conflict.

File: db_funcs.py (hunk 2 above)
Comment (crazylazylife) [reply db_funcs.py:113]: Isn't that like a costly operation for these two simple queries? Like the join implementation took ~1.5 sec to run and the two queries were .5 secs

File: db_funcs.py (hunk 1 above)
Comment (crazylazylife) [reply db_funcs.py:128]: Yep. Updated.

File: db_funcs.py (hunk 4 above)
Comment (crazylazylife) [reply db_funcs.py]: Done.

File: db_funcs.py (hunk 3 above)
Comment (crazylazylife) [reply db_funcs.py:173]: Yes. That was the wrong function update. Initially.

File: db_funcs.py (hunk 2 above)
Comment (crazylazylife) [reply db_funcs.py:115]: Resolving this for now. We can discuss this later if there is any issue.

**Review by DeepankarAcharyya — APPROVED** [review]

**Pull request merged**